src/atom.cpp

```diff
@@ -1884,33 +1884,63 @@ void Atom::allocate_type_arrays()
    called from reading of data file
    type_offset may be used when reading multiple data files
 ------------------------------------------------------------------------- */
+// clang-format on
 
-void Atom::set_mass(const char *file, int line, const char *str, int type_offset,
-                    int labelflag, int *ilabel)
+void Atom::set_mass(const char *file, int line, const char *str, int type_offset, int labelflag,
```

Comment:
on a related note, statements like "Lines can have a trailing comment starting with ‘#’ that is ignored" in the read_data and molecule file docs are not strictly true anymore

Reply:
> also, a more general point, some tools put the trailing '#' without a space after the number, for example:

okay, i remember where i've seen this. it is only in the Dihedral Coeffs of the files produced by msi2lmp. so, really a bug in that code, and currently a nonissue

Reply:
> support for the Masses data file section is nice, though it might open the door for people to ask why other per-type sections aren't also supported (and not sure if that would be worth the effort)

That door was opened by submitting the pull request.  People *will* ask. Even for cases where it won't make sense or is close to impossible to implement. We have to be ready to have a substitution eventually for any place where a (numeric) type is used.

Reply:
> > also, a more general point, some tools put the trailing '#' without a space after the number, for example:
> 
> okay, i remember where i've seen this. it is only in the Dihedral Coeffs of the files produced by msi2lmp. so, really a bug in that code, and currently a nonissue

confirmed and fixed